cdk/handlers/raster_handler.py

```diff
@@ -1,38 +1,82 @@
 """AWS Lambda handler."""
 
 import asyncio
+import json
 import logging
 import os
-
-from eoapi.raster.utils import get_secret_dict
-
-# Update postgres env variables before importing titiler.pgstac.main
-pgstac_secret_arn = os.environ["PGSTAC_SECRET_ARN"]
-pgbouncer_host = os.getenv("PGBOUNCER_HOST")
-
-secret = get_secret_dict(pgstac_secret_arn)
-os.environ.update(
-    {
-        "postgres_host": pgbouncer_host or secret["host"],
-        "postgres_dbname": secret["dbname"],
-        "postgres_user": secret["username"],
-        "postgres_pass": secret["password"],
-        "postgres_port": str(secret["port"]),
-    }
-)
+import re
 
 from eoapi.raster.main import app
+from eoapi.raster.utils import get_secret_dict
+from fastapi import Request
+from fastapi.routing import APIRoute
 from mangum import Mangum
 from titiler.pgstac.db import connect_to_db
+from titiler.pgstac.settings import PostgresSettings
 
 logging.getLogger("mangum.lifespan").setLevel(logging.ERROR)
 logging.getLogger("mangum.http").setLevel(logging.ERROR)
 
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+pgstac_secret_arn = os.environ["PGSTAC_SECRET_ARN"]
+pgbouncer_host = os.getenv("PGBOUNCER_HOST")
+secret = get_secret_dict(pgstac_secret_arn)
+
+pg_settings = PostgresSettings(
+    postgres_host=pgbouncer_host or secret["host"],
+    postgres_dbname=secret["dbname"],
+    postgres_user=secret["username"],
+    postgres_pass=secret["password"],
+    postgres_port=secret["port"],
+)
+
 
 @app.on_event("startup")
 async def startup_event() -> None:
     """Connect to database on startup."""
-    await connect_to_db(app)
+    await connect_to_db(app, settings=pg_settings)
+
+    app.state.path_templates = {}
+    for route in app.routes:
+        if isinstance(route, APIRoute):
+            # replace : with _ to make it regexable
+            route_path = route.path.replace(":", "__")
+            pattern = re.sub(r"{([^}]+)}", r"(?P<\1>[^/]+)", route_path)
+            app.state.path_templates[re.compile(f"^{pattern}$")] = route_path
+
+
+@app.middleware("http")
+async def log_request_data(request: Request, call_next):
+    path = request.url.path
+    method = request.method
+    query_params = dict(request.query_params)
+
+    # Extract path parameters
+    path_template = path
+    path_params = {}
+
+    for pattern, template in app.state.path_templates.items():
+        match = pattern.match(path)
+        if match:
+            path_template = template
+            path_params = match.groupdict()
+            break
+
+    # Build log data
+    log_data = {
+        "method": method,
+        "path_template": path_template,
+        "path": path,
+        "path_params": path_params,
+        "query_params": query_params,
+    }
+
+    logger.info(f"Request: {json.dumps(log_data)}")
+
+    response = await call_next(request)
+    return response
```

Comment:
👍 
maybe we could update https://github.com/developmentseed/titiler/blob/6d1519e9a289a1f1f8e15c04020920f82cf02c60/src/titiler/core/titiler/core/middleware.py#L100-L135 to match what you have here

Reply:
oh that's cool - I didn't know about the existing logging middleware in titiler. yeah, maybe we could expand it a bit with route info.

Reply:
I opened a follow-up https://github.com/developmentseed/titiler/issues/1107